src/App.tsx

```diff
@@ -1,18 +1,49 @@
-import React from 'react';
+import React, {
+  useState, memo, useMemo, useCallback,
+} from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+  .map(numbers => `Item ${numbers}`);
+
+export const App: React.FC = memo(() => {
+  const total = items.length;
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const lastItemOnPage = Math.min(perPage * currentPage, total);
+  const firstItemOnPage = perPage * (currentPage - 1) + 1;
+  const itemsPerPage = useMemo(
+    () => getNumbers(firstItemOnPage, lastItemOnPage),
+    [firstItemOnPage, lastItemOnPage],
+  );
+
+  const handleChangePerPage = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(+event.target.value);
+    setCurrentPage(1);
+  };
+
+  const handlePageChange = useCallback((page: number) => {
+    setCurrentPage(page);
+  }, []);
 
-export const App: React.FC = () => {
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page&nbsp;
+        {currentPage}&nbsp;
+        (
+          items&nbsp;
+        {firstItemOnPage}&nbsp;
+        -&nbsp;
+        {lastItemOnPage}&nbsp;
+        of&nbsp;
+        {total}
+        )
```

Comment:
Please rollback, as it was before)
```suggestion
        {`Page ${currentPage} (items ${firstItemOnPage} - ${lastItemOnPage} of ${total})`}
```